src/rhsm/profile.py

```diff
@@ -59,6 +59,14 @@ def __str__(self):
     def __eq__(self, other):
         return self.content == other.content
 
+    @staticmethod
+    def __uniquify(module_list):
```

Comment:
I don't think it is necessary to have this method "private". I would make this method only "protected": `_uniquify` to make testing easier. I don't see any reason, why this method should not be used in subclasses of `ModulesProfile`.